### DISCUSSION TITLE
How to simulate Dirichlet Boundary?

### DISCUSSION THREAD

**Discussion by ghost** [SeisSol/SeisSol#757] [Q&A]
As title, when I created a surface with boundary type 104 (Dirichlet Boundary), There is an error:

Error: Model for easiBoundary is not initialized!

Then the program raised MPI_ABORT. 

Could you please tell me how to fix this? Thanks!

**Comment by krenzland** [comment]
What boundary conditions do you want to impose? Dirichlet boundary conditions are not well supported at the moment.

**Reply by ghost** [reply]
I want to impose a hard boundary. Is there any way to simulate it?

**Reply by krenzland** [reply]
You mean a zero velocity boundary condition? Which material, elastic or acoustic?

**Reply by ghost** [reply]
Yes, we want to simulate a surface of acoustic material with zero velocity BC.

**Reply by krenzland** [reply]
You need to use this file: https://github.com/SeisSol/Examples/blob/master/convergence_acoustic_elastic/convergence_ocean/rigid.yaml
+ 
this line in parameters.par:
https://github.com/SeisSol/Examples/blob/master/convergence_acoustic_elastic/convergence_ocean/parameters.par#L3
+
this gmsh tag:
https://github.com/SeisSol/Examples/blob/master/convergence_acoustic_elastic/convergence_ocean/mesh_template.geo#L67

Or, in other words: Follow the convergence_ocean example. For this case, the boundary condition is well tested and should work.

**Reply by buzhiqiuyuebai** [reply]
> You need to use this file: https://github.com/SeisSol/Examples/blob/master/convergence_acoustic_elastic/convergence_ocean/rigid.yaml + this line in parameters.par: https://github.com/SeisSol/Examples/blob/master/convergence_acoustic_elastic/convergence_ocean/parameters.par#L3 + this gmsh tag: https://github.com/SeisSol/Examples/blob/master/convergence_acoustic_elastic/convergence_ocean/mesh_template.geo#L67
> 
> Or, in other words: Follow the convergence_ocean example. For this case, the boundary condition is well tested and should work.

May I ask how to achive Dilichret boundary condition by adding 
```
!ConstantMap
    map:
      map_u_u: -1.0 
```
How does this achieve a zero velocity  and what does `map_u_u: -1.0` mean? Does this boundary condition still apply if the scoustic waves travels to the metal? I have read Philipp Wendland‘s article `Time-Reversal of Seismic Waves in SeisSol`, Can the dirichlet boundary condition be achieved by modifying the diagonal matrix  `diag(−1, 1, 1,−1, 1,−1, 1, 1, 1)` on the basis of the free boundary?


![image](https://github.com/SeisSol/SeisSol/assets/114474894/b80caf64-564f-43e7-97a5-5fb74b5c5653)
